src/OpenTelemetry/Trace/ExceptionProcessor.cs

```diff
@@ -19,6 +19,22 @@ internal sealed class ExceptionProcessor : BaseProcessor<Activity>
     public ExceptionProcessor()
     {
 #if NET6_0_OR_GREATER || NETFRAMEWORK
+        bool getExceptionPointersSupported = true;
+
+        try
+        {
+            Marshal.GetExceptionPointers();
+        }
+        catch
+        {
+            getExceptionPointersSupported = false;
+        }
+
+        if (!getExceptionPointersSupported)
+        {
+            throw new PlatformNotSupportedException($"'{typeof(Marshal).FullName}.GetExceptionPointers' is not supported.");
```

Comment:
What does this mean to the user? (e.g. I guess most users would have no idea when they see this exception message, because the method name mentioned here is not what they are using)

Reply:
Please see https://github.com/open-telemetry/opentelemetry-dotnet/pull/5374/files/2708affe2653c55e60e86402602517cd03fd1625#r1508299228